epimodel/exports/epidemics_org.py

```diff
@@ -139,3 +202,140 @@ def types_to_json(obj):
         return obj.name
     else:
         raise TypeError(f"Unserializable object {obj} of type {type(obj)}")
+
+
+def get_cmi(df: pd.DataFrame):
+    return df.index.levels[0].unique()
+
+
+def analyze_data_consistency(
+    debug: Optional[None],
+    export_regions: List[str],
+    models,
+    rates_df,
+    hopkins,
+    foretold,
+) -> None:
+    codes = {
+        "models": get_cmi(models),
+        "hopkins": get_cmi(hopkins),
+        "foretold": get_cmi(foretold),
+        "rates": rates_df.index.unique(),
+    }
+
+    union_codes = set()
+    any_nan = False
+    for source_name, ixs in codes.items():
+        if ixs.isna().sum() > 0:
+            log.error("Dataset %s contains NaN in index!", source_name)
+            any_nan = True
+        union_codes.update(ixs)
+
+    if any_nan:
+        raise ValueError("Some datasets indexed by NaNs. Fix the source data.")
+
+    df = pd.DataFrame(index=sorted(union_codes))
+    for source_name, ixs in codes.items():
+        df[source_name] = pd.Series(True, index=ixs)
+    df = df.fillna(False)
+
+    log.info("Total data availability, number of locations: %s", df.sum().to_dict())
+    log.info("Export requested for %s regions: %s", len(export_regions), export_regions)
+
+    if debug:
+        _df = df.loc[export_regions, ["hopkins", "rates"]]
+        res = _df.loc[~_df.all(axis=1)].replace({False: "Missing", True: "OK"})
+        log.debug(
+            "Data presence for hopkins or rates in the following countries: \n%s", res
+        )
+
+    diff_export_and_models = set(export_regions).difference(get_cmi(models))
+    if diff_export_and_models:
+        log.error(
+            "You requested to export %s but that's not modelled yet.",
+            diff_export_and_models,
+        )
+        raise ValueError(
+            f"Regions {diff_export_and_models} not present in modelled data. Remove it from config."
+        )
+
+    log.info(
+        "From exported regions (N=%s): %s",
+        len(export_regions),
+        df.loc[export_regions].sum().to_dict(),
+    )
+
+
+def get_df_else_none(df: pd.DataFrame, code) -> Optional[pd.DataFrame]:
+    if code in df.index:
+        return df.loc[code].sort_index()
+    else:
+        return None
+
+
+def get_extra_path(args, name: str) -> Path:
+    return Path(args.config["data_dir"]) / args.config["web_export"][name]
+
+
+def aggregate_countries(
```

Comment:
It would be nice to have this part of the epimodel core, somebody already requested or mentioned it, I think?
E.g. having `epimodel.algorithms.sum_column_upwards(rds, df, column)`

Reply:
AFAIK it was mentioned by @mathijshenquet . It's very much hardcoded to hopkins now (e.g. assumes multiindex etc). Do you want me to do this extraction as part of this PR or can that follow up?